### PR TITLE
XW-6 | Integration of Google Universal Analytics with Optimizely

### PR DESCRIPTION
**Pull request by Warkot**
https://wikia-inc.atlassian.net/browse/XW-6

### CODE REVIEW

File: extensions/wikia/AnalyticsEngine/js/universal_analytics.js [hunk 1]
```diff
@@ -5,497 +5,533 @@
...
+			optimizely.activeExperiments &&
+			optimizely.activeExperiments.length > 0 &&
+			typeof optimizely.allExperiments === 'object' &&
+			Object.keys(optimizely.allExperiments).length > 0 &&
+			typeof optimizely.variationNamesMap === 'object' &&
+			Object.keys(optimizely.variationNamesMap).length > 0;
+	}
+
+	if (isOptimizelyLoadedAndActive()) {
+		window.optimizely.activeExperiments.forEach(function (experimentId) {
```
Comment (rogatty) [inline extensions/wikia/AnalyticsEngine/js/universal_analytics.js]: `var optimizely = window.optimizely`

**Comment by rogatty** [comment]
LGTM but can you add somewhere in comments that this logic is the same as in Mercury and put a link to it? If someone will modify this code (s)he should do it in both places.

Jenkins message not related - `/tests/AdminDashboardTest.php` is failing.

**Comment by vforge** [comment]
LGTM

File: extensions/wikia/AnalyticsEngine/js/universal_analytics.js [hunk 2]
```diff
@@ -5,497 +5,536 @@
...
+			Array.isArray(optimizely.activeExperiments) &&
+			optimizely.activeExperiments.length > 0 &&
+			typeof optimizely.allExperiments === 'object' &&
+			Object.keys(optimizely.allExperiments).length > 0 &&
+			typeof optimizely.variationNamesMap === 'object' &&
+			Object.keys(optimizely.variationNamesMap).length > 0;
+	}
+
+	// UA integration code is also used in Mercury SPA - if you change it here, change it there too:
+	// https://github.com/Wikia/mercury/blob/XW-6/front/scripts/mercury/utils/variantTesting.ts#L75
```
Comment (rogatty) [inline extensions/wikia/AnalyticsEngine/js/universal_analytics.js]: This link won't work after deleting the branch. Would be safer too give a link to file in dev branch and a name of the function.

**Pull request merged**